frontends/parsers/p4/p4parser.ypp

```diff
@@ -561,6 +561,60 @@ annotationBody
         }
     ;
 
+annotationName
+    : ABSTRACT         { $$ = new IR::ID(@1, "abstract"_cs); }
```

Comment:
Is there a way to just defer to KEYWORD in yacc? Otherwise we have to manually keep track of this every time we add a new keyword.

Reply:
The issue is the lexer needs to decide what is a keyword without using any contextual info.  We could perhaps change all the keywords to contain the string as a sematic value (so all of these actions would become `{ $$ = new IR::ID(@1, $1); }`, but we'd still need to actually add all the rules in the grammar.

Reply:
I was thinking of a way to generate these rules automatically based on the KEYWORD list, but it looks like there is no way.